public/app/features/alerting/unified/utils/query.ts

```diff
@@ -0,0 +1,66 @@
+import { DataQuery, DataSourceInstanceSettings } from '@grafana/data';
+import { LokiQuery } from 'app/plugins/datasource/loki/types';
+import { PromQuery } from 'app/plugins/datasource/prometheus/types';
+import { CombinedRule } from 'app/types/unified-alerting';
+import { AlertQuery, RulerAlertingRuleDTO } from 'app/types/unified-alerting-dto';
+import { isCloudRulesSource, isGrafanaRulesSource } from './datasource';
+import { isAlertingRulerRule, isGrafanaRulerRule } from './rules';
+
+export function alertRuleToQueries(combinedRule: CombinedRule | undefined | null): AlertQuery[] {
+  if (!combinedRule) {
+    return [];
+  }
+
+  const { namespace, rulerRule } = combinedRule;
+  const { rulesSource } = namespace;
+
+  if (isGrafanaRulesSource(rulesSource)) {
+    if (isGrafanaRulerRule(rulerRule)) {
+      return rulerRule.grafana_alert.data;
+    }
+  }
+
+  if (isCloudRulesSource(rulesSource)) {
+    if (isAlertingRulerRule(rulerRule)) {
```

Comment:
currently it does not show queries for a vanilla prometheus, which only has `promRule`. but `promRule` has a `.query` as well with the expression, we should support it